src/mscorlib/shared/System/Text/StringBuilder.cs

```diff
@@ -2001,6 +2043,22 @@ private static unsafe void ThreadSafeCopy(char* sourcePtr, char[] destination, i
             }
         }
 
+        private static unsafe void ThreadSafeCopy(char* sourcePtr, Span<char> destination, int destinationIndex, int count)
```

Comment:
Do you know for sure that there is a performance penalty? If so, and `Span` is the culprit, we'd love to see examples.